mijia/src/bluetooth/mod.rs

```diff
@@ -329,6 +334,61 @@ pub struct CharacteristicInfo {
     pub id: CharacteristicId,
     /// The 128-bit UUID of the characteristic.
     pub uuid: Uuid,
+    /// The set of flags (a.k.a. properties) of the characteristic, defining how the characteristic
+    /// can be used.
+    pub flags: CharacteristicFlags,
+}
+
+bitflags! {
+    /// The set of flags (a.k.a. properties) of a characteristic, defining how the characteristic
+    /// can be used.
+    pub struct CharacteristicFlags: u16 {
+        const BROADCAST = 0x01;
+        const READ = 0x02;
+        const WRITE_WITHOUT_RESPONSE = 0x04;
+        const WRITE = 0x08;
+        const NOTIFY = 0x10;
+        const INDICATE = 0x20;
+        const SIGNED_WRITE = 0x40;
+        const EXTENDED_PROPERTIES = 0x80;
+        const RELIABLE_WRITE = 0x100;
+        const WRITABLE_AUXILIARIES = 0x200;
+        const ENCRYPT_READ = 0x400;
+        const ENCRYPT_WRITE = 0x800;
+        const ENCRYPT_AUTHENTICATED_READ = 0x1000;
+        const ENCRYPT_AUTHENTICATED_WRITE = 0x2000;
+        const AUTHORIZE = 0x4000;
```

Comment:
do these values happen to be the same ones that the bluetooth spec uses?

Reply:
The first 10 are at least. Or really, the first 8 are what goes in the properties field, `RELIABLE_WRITE` and `WRITABLE_AUXILIARIES` go in the extended properties descriptor. The others don't seem to be mentioned in the spec as far as I can see, but are listed in the BlueZ documentation so I figured we should probably include them.